filament/backend/src/SamplerGroup.cpp

```diff
@@ -62,13 +62,10 @@ SamplerGroup& SamplerGroup::setSamplers(SamplerGroup const& rhs) noexcept {
     return *this;
 }
 
-void SamplerGroup::setSampler(size_t index, Sampler const& sampler) noexcept {
-    if (index < mBuffer.size()) {
-        auto& cur = mBuffer[index];
-        if (cur.t != sampler.t || cur.s.u != sampler.s.u) {
-            cur = sampler;
-            mDirty.set(index);
-        }
+void SamplerGroup::setSampler(size_t index, Sampler sampler) noexcept {
+    if (UTILS_LIKELY(index < mBuffer.size())) {
```

Comment:
could you put a comment in the code that we can't compare Handle<> and also a TODO we need to find a way to not mark dirty if the texture doesn't actually change